DESIGN.md

```diff
@@ -0,0 +1,169 @@
+
+For example, a new logging component could be added in multiple possible ways to Warnet:
+
+1. Run a process on the local host which connects in to the cluster via a forwarded port and performs customised logging.
+
+2. Launch a standardised logging application (e.g. Grafana) via executing a `kubectl` command on a .yaml file.
+
+  ```python
+  subprocess.run("kubectl apply -f grafana.yaml --namespace=grafana")
+  ```
+
+3. Create a helm chart (or use an [already-available](https://github.com/grafana/helm-charts) one) for the Grafana component and install using helm.
+
+  ```python
+  subprocess.run("helm upgrade --install --namespace logging promtail grafana/promtail")
+  ```
+
+Out of these three options, the third should be preferred where possible, followed by the second, with the first only being used in extreme cases.
+
+## Kubernetes and Helm in Warnet
+
+<div style="display: flex; justify-content: space-around; align-items: center; margin: 20px 0;">
+  <img src="/img/kubernetes.svg" alt="Kubernetes logo" style="width: 100px; height: 100px;" />
+  <img src="/img/helm.svg" alt="Helm logo" style="width: 100px; height: 100px;" />
+</div>
+
+Warnet leverages [Kubernetes](https://kubernetes.io/) for deploying and managing simulated Bitcoin networks, with [Helm](https://helm.sh/) serving as the preferred method for managing Kubernetes resources. Understanding the relationship between these technologies is crucial for grasping Warnet's architecture and deployment strategy.
+
+### Kubernetes
+[Kubernetes](https://kubernetes.io/) is an open-source container orchestration platform that automates the deployment, scaling, and management of containerized applications. In Warnet, Kubernetes provides the underlying infrastructure for running simulated Bitcoin nodes and related services.
+
+Key benefits of using Kubernetes in Warnet:
+- Scalability: Easily scale the number of simulated nodes
+- Resource management: Efficiently allocate computational resources
+- Service discovery: Automatically manage network connections between nodes
+
+### Helm
+[Helm](https://helm.sh/) is a package manager for Kubernetes that simplifies the process of defining, installing, and upgrading complex Kubernetes applications. Warnet prefers Helm for managing Kubernetes resources due to its powerful templating and package management capabilities.
+
+Advantages of using Helm in Warnet:
+- Templating: Define reusable Kubernetes resource templates
+- Versioning: Manage different versions of deployed resources
+- Packaging: Bundle related Kubernetes resources into a single unit (chart)
+- Simplified deployment: Use a single command to deploy complex applications
+
+### Why Helm is Preferred in Warnet
+1. Reproducibility: Helm charts ensure consistent deployments across different environments.
```

Comment:
I think this one might be debatable as wouldn't we get reproducibility writing yaml manually with python?

Reply:
I can remove, although I think we could make lots of things reproducible, perhaps even bash scripts!

This does make me think though, perhaps I've missed the primary helm advantage, in the way that is merges configurations together. Is there a technical term for this do you know?

Reply:
Can't think of a technical term but for me helm is great for its, templating, configuration inheritance, encapsulation / modularity.